src/realm/sync/subscriptions.cpp

```diff
@@ -630,22 +630,26 @@ SubscriptionStore::SubscriptionStore(Private, DBRef db)
     };
 
     auto tr = m_db->start_read();
-    SyncMetadataSchemaVersions schema_versions(tr);
+    // Start with a reader so it doesn't try to write until we are ready
+    SyncMetadataSchemaVersionsReader schema_versions_reader(tr);
 
-    if (auto schema_version = schema_versions.get_version_for(tr, internal_schema_groups::c_flx_subscription_store);
-        !schema_version) {
-        tr->promote_to_write();
-        schema_versions.set_version_for(tr, internal_schema_groups::c_flx_subscription_store, c_flx_schema_version);
-        create_sync_metadata_schema(tr, &internal_tables);
-        tr->commit_and_continue_as_read();
-    }
-    else {
+    if (auto schema_version =
+            schema_versions_reader.get_version_for(tr, internal_schema_groups::c_flx_subscription_store)) {
         if (*schema_version != c_flx_schema_version) {
             throw RuntimeError(ErrorCodes::UnsupportedFileFormatVersion,
                                "Invalid schema version for flexible sync metadata");
         }
         load_sync_metadata_schema(tr, &internal_tables);
     }
+    else {
+        tr->promote_to_write();
+        // Create the metadata schema
+        create_sync_metadata_schema(tr, &internal_tables);
```

Comment:
I had originally swapped them to ensure the schema was written before the schema version, but, since they are in the same commit, it really doesn't matter. I also made some updates around this area so the initial creation of the schema versions table is in its own commit.